x-pack/plugins/fleet/server/services/agents/action_runner.ts

```diff
@@ -113,6 +113,13 @@ export abstract class ActionRunner {
             if (this.retryParams.retryCount === 3) {
               const errorMessage = 'Stopping after 3rd retry. Error: ' + error.message;
               appContextService.getLogger().warn(errorMessage);
+
+              // no need for check task, action failed
+              this.bulkActionsResolver!.removeIfExists(this.checkTaskId!);
```

Comment:
Looks like this methods are async should we not wrap them in a `await Promise.all()`?